app-commons/src/main/scala/org/bitcoins/commons/jsonmodels/dlc/DLCMessage.scala

```diff
@@ -42,64 +44,382 @@ object DLCMessage {
-        case (vec, (digest, sats)) => vec ++ digest.bytes ++ sats.bytes
+        case (vec, (str, sats)) =>
+          val strBytes = CryptoUtil.serializeForHash(str.outcome)
+          vec ++ BigSizeUInt.calcFor(strBytes).bytes ++ strBytes ++ sats.bytes
       }
     }
 
-    def toTLV: ContractInfoV0TLV = ContractInfoV0TLV(outcomeValueMap)
-  }
+    override def toTLV: ContractInfoV0TLV =
+      ContractInfoV0TLV(outcomeValueMap.map {
+        case (outcome, amt) => outcome.outcome -> amt
+      })
 
-  object ContractInfo extends Factory[ContractInfo] {
+    override def flip(totalCollateral: Satoshis): SingleNonceContractInfo = {
+      SingleNonceContractInfo(outcomeValueMap.map {
+        case (hash, amt) => (hash, (totalCollateral - amt).satoshis)
+      })
+    }
+  }
 
-    private val sizeOfMapElement: Int = 40
+  object SingleNonceContractInfo extends Factory[SingleNonceContractInfo] {
 
-    val empty: ContractInfo = ContractInfo(ByteVector.low(sizeOfMapElement))
+    def fromStringMap(map: Map[String, Satoshis]): SingleNonceContractInfo = {
+      SingleNonceContractInfo(map.map {
+        case (str, amt) => EnumOutcome(str) -> amt
+      })
+    }
 
-    override def fromBytes(bytes: ByteVector): ContractInfo = {
+    override def fromBytes(bytes: ByteVector): SingleNonceContractInfo = {
       @tailrec
       def loop(
           remainingBytes: ByteVector,
-          accum: Vector[(Sha256Digest, Satoshis)]): Vector[
-        (Sha256Digest, Satoshis)] = {
-        if (remainingBytes.size < sizeOfMapElement) {
+          accum: Vector[(EnumOutcome, Satoshis)]): Vector[
+        (EnumOutcome, Satoshis)] = {
+        if (remainingBytes.isEmpty) {
           accum
         } else {
-          val relevantBytes = remainingBytes.take(sizeOfMapElement)
-          val digest = Sha256Digest(relevantBytes.take(32))
-          val sats = Satoshis(relevantBytes.takeRight(8))
-          loop(remainingBytes.drop(sizeOfMapElement), accum :+ (digest, sats))
+          val outcomeSize = BigSizeUInt(remainingBytes)
+          val outcome =
+            remainingBytes.drop(outcomeSize.byteSize).take(outcomeSize.toInt)
+          val outcomeStr = new String(outcome.toArray, StandardCharsets.UTF_8)
+          val outcomeSizeAndOutcomeLen =
+            outcomeSize.byteSize + outcomeSize.toInt
+          val sats = Satoshis(
+            remainingBytes.drop(outcomeSizeAndOutcomeLen).take(8))
+
+          loop(remainingBytes.drop(outcomeSizeAndOutcomeLen + 8),
+               accum :+ (EnumOutcome(outcomeStr), sats))
         }
       }
-      ContractInfo(loop(bytes, Vector.empty).toMap)
+      SingleNonceContractInfo(loop(bytes, Vector.empty).toMap)
+    }
+  }
+
+  case class MultiNonceContractInfo(
+      outcomeValueFunc: OutcomeValueFunction,
+      base: Int,
+      numDigits: Int,
+      totalCollateral: Satoshis)
+      extends ContractInfo {
+
+    lazy val outcomeVec: Vector[(Vector[Int], Satoshis)] =
```

Comment:
consider making non-lazy in case this makes things bad when parallelism happens?